### PR TITLE
feat(rds): enhanced monitoring configuration at the cluster level

### PR DESCRIPTION
**Pull request by badmintoncryer**
### Issue # (if applicable)

None

### Reason for this change

AWS RDS now supports the configuration of advanced monitoring at the cluster level.

<https://aws.amazon.com/about-aws/whats-new/2024/09/amazon-rds-performance-insights-supports-aurora-cluster-level-configuration/?nc1=h_ls>

This feature is essential for creating limitless database cluster. (#32151)

### Description of changes

I added `enableClusterLevelEnhancedMonitoring` prop to `ClusterProps`.
If this prop enabled, `monitoringInterval` and `monitoringRole` is applied to not the instances but the cluster.

Enhanced monitoring configuration for the cluster and instances has some restrictions.
- When setting the `monitoringRoleArn` for both the cluster and instances, the values must be identical.
- When setting the `monitoringInterval` for both the cluster and instances, the values must be identical.
- When the `monitoringInterval` or `monitoringRoleArn` is set at the cluster level, enhanced monitoring will be enabled even if the same values are not specified for the instances.

Based on the above, I decided to add a flag (`enableClusterLevelEnhancedMonitoring`) to switch the application of `monitoringInterval` and `monitoringRole` between the instance level and the cluster level to avoid breaking changes to the existing props.

#### Verification Results Memo

The value of the Enhanced Monitoring configuration must be the same for both the cluster level and the instance level.

```ts
  "DatabaseB269D8BB": {
   "Type": "AWS::RDS::DBCluster",
   "Properties": {
    ...,
    "Engine": "aurora-postgresql",
    "EngineVersion": "16.1",
    "MonitoringInterval": 5,
    "MonitoringRoleArn": {
     "Fn::GetAtt": [
      "DatabaseMonitoringRoleForCluster91F5067E",
      "Arn"
     ]
    },
  },
  "DatabasewriterInstanceEBFCC003": {
   "Type": "AWS::RDS::DBInstance",
   "Properties": {
    ...,
    "DBClusterIdentifier": {
     "Ref": "DatabaseB269D8BB"
    },
    "Engine": "aurora-postgresql",
    "MonitoringInterval": 1,
    "MonitoringRoleArn": {
     "Fn::GetAtt": [
      "DatabaseMonitoringRole576991DA",
      "Arn"
     ]
    },
  }
```

```sh
           | Database/writerInstance (DatabasewriterInstanceEBFCC003) Resource handler returned message: "MonitoringInterval conflicts with cluster level parameter. (Service: Rds, Status Code: 400, Request ID: 7c9d9023-fe2a-4b39-a939-22cec3595041)" (RequestToken: bc3e18eb-2daa-4f25-c205-11f1ec70fc15, HandlerErrorCode: InvalidRequest)
```

The monitoring role arn must be the same for both cluster and instance level.

```yaml
 "DatabaseB269D8BB": {
   "Type": "AWS::RDS::DBCluster",
   "Properties": {
    ...,
    "Engine": "aurora-postgresql",
    "EngineVersion": "16.1",
    "MonitoringInterval": 5,
    "MonitoringRoleArn": {
     "Fn::GetAtt": [
      "DatabaseMonitoringRoleForCluster91F5067E",
      "Arn"
     ]
    },
  },
  "DatabasewriterInstanceEBFCC003": {
   "Type": "AWS::RDS::DBInstance",
   "Properties": {
    ...,
    "DBClusterIdentifier": {
     "Ref": "DatabaseB269D8BB"
    },
    "Engine": "aurora-postgresql",
    "MonitoringInterval": 5,
    "MonitoringRoleArn": {
     "Fn::GetAtt": [
      "DatabaseMonitoringRole576991DA",
      "Arn"
     ]
    },
  }
```

```sh
Resource handler returned message: "MonitoringRoleArn conflicts with cluster level parameter. (Service: Rds, Status Code: 400, Request ID: 761a867d-409e-4208-a08e-5591607d0ed0)" (RequestToken: fb691324-f37c-405d-e5bd-447911f7664f, HandlerErrorCode: InvalidRequest)
```

### Description of how you validated changes

Add both unit and integ tests

### Checklist
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/aws-cdk-lib/aws-rds/lib/cluster.ts [hunk 1]
```diff
@@ -811,25 +851,15 @@ abstract class DatabaseClusterNew extends DatabaseClusterBase {
...
-        ],
-      });
-    }
-
     // need to create the writer first since writer is determined by what instance is first
     const writer = props.writer!.bind(this, this, {
-      monitoringInterval: props.monitoringInterval,
-      monitoringRole: monitoringRole,
+      monitoringInterval: props.enableClusterLevelEnhancedMonitoring ? undefined : props.monitoringInterval,
+      monitoringRole: props.enableClusterLevelEnhancedMonitoring ? undefined : this.monitoringRole,
```
Comment (GavinZZ) [inline packages/aws-cdk-lib/aws-rds/lib/cluster.ts:862]: Would be nice to have a comment to explain that when `enableClusterLevelEnhancedMonitoring` is enabled, we already set interval and role at cluster level so no need to re-set it in instance level.

File: packages/@aws-cdk-testing/framework-integ/test/aws-rds/test/integ.temp.js.snapshot/EnhancedMonitoringClusterTestStack.template.json [hunk 2]
```diff
@@ -0,0 +1,858 @@
...
+    "CidrBlock": "10.0.0.0/16",
+    "EnableDnsHostnames": true,
+    "EnableDnsSupport": true,
+    "InstanceTenancy": "default",
+    "Tags": [
+     {
+      "Key": "Name",
+      "Value": "EnhancedMonitoringClusterTestStack/Vpc"
+     }
+    ]
```
Comment (GavinZZ) [inline packages/@aws-cdk-testing/framework-integ/test/aws-rds/test/integ.temp.js.snapshot/EnhancedMonitoringClusterTestStack.template.json:15]: what is this `integ.temp.js` test? I don't see any integration testss attach to it.

File: packages/aws-cdk-lib/aws-rds/README.md [hunk 3]
```diff
@@ -1372,3 +1372,48 @@ To see Amazon RDS DB engines that support Performance Insights, see [Amazon RDS
 To see Amazon Aurora DB engines that support Performance Insights, see [Amazon Aurora DB engine, Region, and instance class support for Performance Insights](https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/USER_PerfInsights.Overview.Engines.html).
 
 For more information about Performance Insights, see [Monitoring DB load with Performance Insights on Amazon RDS](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/USER_PerfInsights.html).
+
+## Enhanced Monitoring
+
+With [Enhanced Monitoring](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/USER_Monitoring.OS.html#USER_Monitoring.OS.Enabling), you can monitor the operating system of your DB instance in real time.
+
+To enable Enhanced Monitoring for a clustered database, set the `monitoringInterval` property.
+This value is applied to all instances in the cluster by default.
```
Comment (GavinZZ) [inline packages/aws-cdk-lib/aws-rds/README.md:1381]:
```suggestion
This value is applied at instance level to all instances in the cluster by default.
```

File: packages/aws-cdk-lib/aws-rds/README.md [hunk 4]
```diff
@@ -1372,3 +1372,48 @@ To see Amazon RDS DB engines that support Performance Insights, see [Amazon RDS
...
+
+If you want to enable enhanced monitoring at the cluster level, you can set the `enableClusterLevelEnhancedMonitoring` property to `true`.
```
Comment (GavinZZ) [inline packages/aws-cdk-lib/aws-rds/README.md:1383]:
```suggestion
If you want to enable enhanced monitoring at the cluster level, you can set the `enableClusterLevelEnhancedMonitoring` property to `true`. Note that you must set `monitoringInterval` when using `enableClusterLevelEnhancedMonitoring`
```

**Review by GavinZZ — CHANGES_REQUESTED** [review]
Left some feedback, thanks for the contribution!

File: packages/@aws-cdk-testing/framework-integ/test/aws-rds/test/integ.temp.js.snapshot/EnhancedMonitoringClusterTestStack.template.json (hunk 2 above)
Comment (badmintoncryer) [reply packages/@aws-cdk-testing/framework-integ/test/aws-rds/test/integ.temp.js.snapshot/EnhancedMonitoringClusterTestStack.template.json:15]: I'm sorry.. I've removed it.

File: packages/aws-cdk-lib/aws-rds/README.md [hunk 5]
```diff
@@ -1372,3 +1372,48 @@ To see Amazon RDS DB engines that support Performance Insights, see [Amazon RDS
...
+
+```ts
+declare const vpc: ec2.Vpc;
+// Enable Enhanced Monitoring for all of the instances in the cluster
+new rds.DatabaseCluster(this, 'Cluster', {
+  engine: rds.DatabaseClusterEngine.auroraPostgres({ version: rds.AuroraPostgresEngineVersion.VER_16_1 }),
+  writer: rds.ClusterInstance.serverlessV2('writerInstance'),
+  vpc,
+  monitoringInterval: Duration.seconds(5),
+});
```
[the remark this answers is not in the document]
Comment (badmintoncryer) [reply packages/aws-cdk-lib/aws-rds/README.md:1393]: This code block is an example of configuring enhanced monitoring at the instance level.

```md
// Enable Enhanced Monitoring at instance level to all instances in the cluster
new rds.DatabaseCluster(this, 'Cluster', {
  engine: rds.DatabaseClusterEngine.auroraPostgres({ version: rds.AuroraPostgresEngineVersion.VER_16_1 }),
  writer: rds.ClusterInstance.serverlessV2('writerInstance'),
  vpc,
  monitoringInterval: Duration.seconds(5),
});

// Enable Enhanced Monitoring at the cluster level
new rds.DatabaseCluster(this, 'Cluster', {
  engine: rds.DatabaseClusterEngine.auroraPostgres({ version: rds.AuroraPostgresEngineVersion.VER_16_1 }),
  writer: rds.ClusterInstance.serverlessV2('writerInstance'),
  vpc,
  monitoringInterval: Duration.seconds(5),
  enableClusterLevelEnhancedMonitoring: true,
});
```

File: packages/aws-cdk-lib/aws-rds/lib/cluster.ts (hunk 1 above)
Comment (badmintoncryer) [reply packages/aws-cdk-lib/aws-rds/lib/cluster.ts:864]: I have added such explanations to all the sections where instance-level enhanced monitoring settings are applied.

File: packages/aws-cdk-lib/aws-rds/README.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (GavinZZ) [reply packages/aws-cdk-lib/aws-rds/README.md:1393]: I see! The text above got me confused `If you want to enable enhanced monitoring at the cluster level` and I didn't notice the comment you. Looks good.

**Review by GavinZZ — APPROVED** [review]

File: packages/aws-cdk-lib/aws-rds/README.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (badmintoncryer) [reply packages/aws-cdk-lib/aws-rds/README.md:1393]: No problem! Thank you for your confirmation!

**Pull request merged**